InboundHAProxyHandler is not @Sharable

```
Caused by: io.netty.channel.ChannelPipelineException: org.apache.cassandra.stargate.transport.internal.HAProxyProtocolDetectingDecoder$InboundHAProxyHandler is not a @Sharable handler, so can't be added or removed multiple times.
	at io.netty.channel.DefaultChannelPipeline.checkMultiplicity(DefaultChannelPipeline.java:600)
	at io.netty.channel.DefaultChannelPipeline.addAfter(DefaultChannelPipeline.java:300)
	at io.netty.channel.DefaultChannelPipeline.addAfter(DefaultChannelPipeline.java:290)
	at org.apache.cassandra.stargate.transport.internal.HAProxyProtocolDetectingDecoder.decode(HAProxyProtocolDetectingDecoder.java:76)
```
	

I'll submit a PR shortly...

--- a/cql/src/main/java/org/apache/cassandra/stargate/transport/internal/HAProxyProtocolDetectingDecoder.java
+++ b/cql/src/main/java/org/apache/cassandra/stargate/transport/internal/HAProxyProtocolDetectingDecoder.java
@@ -75,6 +75,7 @@ public class HAProxyProtocolDetectingDecoder extends ByteToMessageDecoder {
     return true;
   }
 
+  @Sharable
   static class InboundHAProxyHandler extends SimpleChannelInboundHandler<HAProxyMessage> {
     public static final InboundHAProxyHandler INSTANCE = new InboundHAProxyHandler();
 

